modules/Emails/EmailsController.php

```diff
@@ -639,4 +673,78 @@ protected function setAfterImport($importedEmailId, $request)
 
         return $emails;
     }
+
+    /**
+     * @param User $requestedUser
+     * @param InboundEmail $requestedInboundEmail
+     * @param Email $requestedEmail
+     * @return bool false if user doesn't have access
+     */
+    protected function userIsAllowedToSendEmail($requestedUser, $requestedInboundEmail, $requestedEmail)
+    {
+        $hasAccess = false;
+
+        // Check that user is allowed to use inbound email account
+        $hasAccessToInboundEmailAccount = false;
+        $usersInboundEmailAccounts = $requestedInboundEmail->retrieveAllByGroupIdWithGroupAccounts($requestedUser->id);
+        foreach ($usersInboundEmailAccounts as $inboundEmailId => $userInboundEmail) {
+            if ($userInboundEmail->id === $requestedInboundEmail->id) {
+                $hasAccessToInboundEmailAccount = true;
+                break;
+            }
+        }
+
+        $inboundEmailStoredOptions = $requestedInboundEmail->getStoredOptions();
+
+        // if group email account check that user is allowed to use email account
+        if ($requestedInboundEmail->isGroupEmailAccount()) {
+            if ($inboundEmailStoredOptions['allow_outbound_group_usage'] === true) {
+                $hasAccessToInboundEmailAccount = true;
```

Comment:
@daniel-samson I don't understand what you're trying to say. Can you elaborate please?

When I say that `retrieveAllByGroupId`/`retrieveAllByGroupIdWithGroupAccounts` are broken, is because they clearly do not do what the documentation says it does.

It says:
> Retrieves an array of I-E beans that the user has team access to including group

But it does:

Retrieves an array of I-E beans with all the group email accounts and (optionally) the personal accounts the user has access to.